bindings/matrix-sdk-ffi/src/room_directory_search.rs

```diff
@@ -87,9 +88,15 @@ impl RoomDirectorySearch {
         Ok(())
     }
 
-    pub async fn search(&self, filter: Option<String>, batch_size: u32) -> Result<(), ClientError> {
+    pub async fn search(
+        &self,
+        filter: Option<String>,
+        batch_size: u32,
+        server_name: Option<String>,
```

Comment:
can we call this "via_server_name"? a cursory look made me think it was a server name we were looking for (before i saw `filter`).